File: JS/8/example8.js
#!/usr/bin/env node

'use strict';

const people = require('./people');

people.people.forEach( function (person) {

    console.log("Person ...");
    var yearsLeftTilRetire = 0;

    Object.keys(person).forEach( function(key) {

        console.log(key + ": " + person[key]);

        if (key === 'age')
        {
            var age;
            if (typeof person['age'] === 'string')
            {
                age = parseInt(person['age']);
            }
            else if (typeof person['age'] === 'number')
            {
                age = person['age'];
            }
            yearsLeftTilRetire = 65 - age;
        }

    });

    console.log("Years left: " + yearsLeftTilRetire);
    console.log("");
});

File: Java/basic/src/main/java/org/prodigius/javabasic/example8/Example8.java
package org.prodigius.javabasic.example8;

import java.util.*;

public class Example8
{
    public static void main(String[] args)
    {
        People people = new People();
        people.build();

        for (Map<String,Object> person : people.getPeople())
        {
            int yearsLeft;

            try
            {
                yearsLeft = 65 - (int) person.get("age");
            }
            catch(ClassCastException e)
            {
                yearsLeft = 65 - Integer.parseInt(
                    person.get("age").toString());
            }

            System.out.println("Person ...");
            System.out.println("Name: " + person.get("name"));
            System.out.println("Age: " + person.get("age"));
            System.out.println("Hair: " + person.get("hair"));
            System.out.println("Years Left: " + yearsLeft);
            System.out.println("");
        }
    }
}